### DISCUSSION TITLE
noahprince22's tractyl-manuform-keyboard

### DISCUSSION THREAD

**Discussion by tandy-1000** [joshreve/dactyl-keyboard#43] [Ideas]
I'm closely following the work on the modularisation of the dactyl, mainly the addition of the trackball, as this would be the ideal all in one input device.

I was initially planning to make myself a [tractyl manuform](https://github.com/noahprince22/tractyl-manuform-keyboard), but I realised that you could only make a 4x5 case.

The 'ideal' dactyl manuform would be:
 - 5x6,
 - include a trackball on one half,
 - include a rotary encoder for scrolling on another half (EVQWGD001 like [this](https://old.reddit.com/r/olkb/comments/o7skep/true_wireless_split_keyboard_with_ink_display/))

This project seems to make this much more possible.

My main question is whether there are plans to implement a trackball in a more similar style to @noahprince22?
I know that a PR was merged recently adding a trackball thumb cluster, but I also read that @joshreve was planning their own implementation, so I was wondering what that would look like.

**Comment by joshreve** [comment]
That is me and the trackball mentioned is the Orbyl design I'm currently debugging.  The trackball geometry and generator that incorporates it with any key config is up and running, but I am just starting printing and testing.  I have two thumb cluster trackballs, orbyl and one added by another user and a trackball in wall design [that was just posted on reddit](https://www.reddit.com/r/ErgoMechKeyboards/comments/po837t/douballz_dual_tractyl_build/). 

The specific design for Tractyl doesn't look like it is compatible with a generator which matches what your statement on the 4X5 only, but I haven't looked at it in a ton of detail.

The current missing step on my setup is firmware and a controller capable of OLED and trackball, which [u/tenstaana](https://www.reddit.com/user/tenstaana/) solved with QMK and an elite C to get sufficient I/O. 

**What specifically are you looking for?**

**Reply by tandy-1000** [reply]
I see, thanks for the clarification, and thank you for all the work you're doing!
I was wanted to know if there was any plan to incorporate a potential option to enable a design like the tractyl? It looks to be the most comfortable of dactyls with trackballs, but it's understandable if this is not possible because of your aim for a generatable design.
I was also going to ask if you had any plans to implement an option for a scroll wheel, based on something like the EVQWGD001?

**Comment by joshreve** [comment]
Ah, I see, you liked the placement.  For the tenting, you can absolutely use the generator to get a more aggressive angle between the thumb and finger sections.  It takes a bit of tweaking to get it right, but very doable.  The ball placement is a challenge.  That is on the separation of the cluster and the wall, making it difficult to cleanly generate.  The current designs could either pull down the left wall a bit and move the ball with it, or move the cluster as close as possible to the left wall, but it would have to stop just short of the wall to generate well.

The pulled wall and trackball in wall can be done with just parameters.  I don't think it would fully get to that design, but I think it could get close.  What thumb cluster are you looking to use it with?

On the scroll wheel or encoder, I have no problem including them, but am not actively doing so at the moment.  If someone creates the geometry generation function using the geometry helper functions or even as simple FreeCAD model, I can add the code to place it.  The debugging of the hardware fit is really the larger amount of work and I don't have hardware on hand or time to focus on that at the moment.  If you are handy with FreeCAD and create the interface geometry, I can add parameters to place it in the generator.

**Reply by joshreve** [reply]
![image](https://user-images.githubusercontent.com/5843736/133891601-44643e1b-36e9-4685-9405-a3a6620b76eb.png)

Here is the example of how much you can shift the trackball in wall.  You can also go the other way with the thumb cluster.  You can smash them pretty close together, it just gets ugly when any significant portion of the geometry crosses into another "region".

**Reply by tandy-1000** [reply]
Thank you for all the info! I realise now that the tractyl is quite tented (which allows the in-wall trackball) making it quite different to regular DacMan designs.

> What thumb cluster are you looking to use it with?

I'm not actually sure, I know there are a few available, but I haven't been able to find stls / images of them to decide..

>  If you are handy with FreeCAD and create the interface geometry

Sadly I am not, but I am very interested in it, if you have a matrix chat then I would be happy to contribute with other tasks that need doing to ease the workload :).

The model that you have sent looks quite good already!

**Reply by joshreve** [reply]
I should probably start a chat for people to discuss and share config files.  I'll have to look at adding a discord channel or something.  Right now everything is in github with occasional posts to Reddit.

I also want to create a gallery for the configs to show what the options are.  I'm slowly working on it, but it will take a while.

**Reply by tandy-1000** [reply]
I would suggest a Matrix space since its an open platform, is bridged to gitter.im, can be bridged to discord nicely with t2bot.io. It's more developer oriented :) If you need a hand with making the space, send me a message on Matrix ((at)tandy1000:matrix.org).
